nixos/modules/services/hardware/usbipd.nix

```diff
@@ -0,0 +1,99 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  cfg = config.services.usbipd;
+  device = lib.types.submodule {
+    options = {
+      productid = lib.mkOption {
+        type = lib.types.str;
+        description = "The product id of the device";
+      };
+      vendorid = lib.mkOption {
+        type = lib.types.str;
+        description = "The vendor id of the device";
+      };
+    };
+  };
+in
+{
+  options.services.usbipd = {
+    enable = lib.mkEnableOption "usbip server";
+    kernelPackage = lib.mkPackageOption pkgs.linuxPackages_latest "usbip" { };
+    devices = lib.mkOption {
+      type = lib.types.listOf device;
+      default = [ ];
+      description = "List of USB devices to watch and automatically export.";
+      example = [
+        {
+          productid = "xxxx";
+          vendorid = "xxxx";
+        }
+      ];
+    };
+    openFirewall = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = "Open port 3240 for usbipd";
+      example = false;
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    boot.extraModulePackages = [ cfg.kernelPackage ];
+    boot.kernelModules = [
+      "usbip-core"
+      "usbip-host"
+    ];
+    networking.firewall.allowedTCPPorts = lib.mkIf cfg.openFirewall [ 3240 ];
+    services.udev.extraRules = lib.strings.concatLines (
+      (map (
+        dev:
+        "ACTION==\"add\", SUBSYSTEM==\"usb\", ATTRS{idProduct}==\"${dev.productid}\", ATTRS{idVendor}==\"${dev.vendorid}\", RUN+=\"${pkgs.systemd}/bin/systemctl restart usbip-${dev.vendorid}:${dev.productid}.service\""
```

Comment:
I think this will be more readable by using double single quotes for the overall string, so that the inner double quotes doesn't have to be escaped.